dspace-uclouvain/src/main/java/org/dspace/uclouvain/pdfAttestationGenerator/rest/PDFAttestationGeneratorRestController.java

```diff
@@ -0,0 +1,87 @@
+import org.dspace.uclouvain.pdfAttestationGenerator.exceptions.PDFGenerationException;
+import org.dspace.uclouvain.pdfAttestationGenerator.factory.PDFAttestationGeneratorFactory;
+import org.dspace.uclouvain.pdfAttestationGenerator.handlers.PDFAttestationGeneratorHandler;
+import org.dspace.web.ContextUtil;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+import org.xml.sax.SAXException;
+
+@RestController
+@RequestMapping("/api/uclouvain/item/{uuid}/attestation")
+public class PDFAttestationGeneratorRestController {
+
+    @Autowired
+    private ItemService itemService;
+
+    @Autowired
+    private AuthorizeService authorizeService;
+
+    /** 
+     * Generates and returns a PDF attestation with a template depending on the targeted DSpace object type
+     * 
+     * @param uuid: The uuid of the target DSpace Item from the request
+     */
+    @RequestMapping(method = RequestMethod.GET)
+    public void attestation(
+        HttpServletResponse response,
+        HttpServletRequest request,
+        @PathVariable UUID uuid
+    ) throws FileNotFoundException, IOException, FOPException, TransformerConfigurationException, TransformerException, SAXException, SQLException {
+        try {
+            PDFAttestationGeneratorHandler handler = PDFAttestationGeneratorFactory.getInstance().getHandlerInstance(uuid);
+            if (handler == null) {
+                response.sendError(400, "Object not found or not handled");
+            }
+            else if(this.checkAuthorization(request, uuid)) {
+                try {
+                    response.setContentType("application/pdf");
+                    handler.getAttestation(response.getOutputStream(), uuid);
+                    response.flushBuffer();
+                } catch (PDFGenerationException e) {
+                    System.out.println(e.getMessage());
+                    response.sendError(500, "An error occurred while generating the attestation");
+                }
+            }
+            else {
+                response.sendError(404);
+            }
+        } catch (Exception e) {
+            System.out.println(e);
+            throw new RuntimeException(e);
+        }
+    }
+
+    private Boolean checkAuthorization(HttpServletRequest request, UUID uuid) throws SQLException {
+        Context ctx = ContextUtil.obtainContext(request);
+        EPerson currentUser = ctx.getCurrentUser();
+        Item dsItem = itemService.find(ctx, uuid);
+        if (currentUser != null) {
+            if (authorizeService.isAdmin(ctx, dsItem) || (dsItem.getSubmitter() == ctx.getCurrentUser())) {
+                return true;
+            }
+            return false;
+        }
+        return false;
```

Comment:
try to minimize `if` statement imbrication to be more readable (and remove a lot of spaces)
```java
if (currentUser == null) return false;
return authorizeService.isAdmin(ctx, dsItem) || dsItem.getSubmitter() == ctx.getCurrentUser();
```
saving 5 code lines